.github/workflows/ci.yaml

```diff
@@ -79,3 +79,27 @@ jobs:
     uses: bakdata/ci-templates/.github/workflows/python-poetry-publish-snapshot.yaml@v1.25.0
     secrets:
       pypi-token: ${{ secrets.TEST_PYPI_TOKEN }}
+
+  publish-docs-to-dev:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3   
+        with:
+          ref: ${{ github.head_ref }}
```

Comment:
why?

Reply:
@disrupted This is the error I got: https://github.com/bakdata/kpops/actions/runs/5961286480/job/16170206761?pr=339#step:3:170

And here is the suggested solution:
https://github.com/actions/checkout/issues/317